src/main/kotlin/org/rust/lang/core/macros/MacroExpander.kt

```diff
@@ -239,7 +239,13 @@ private class MacroPattern private constructor(
                         MacroExpansionMarks.failMatchPatternByBindingType.hit()
                         return null
                     }
-                    map[name] = macroCallBody.originalText.substring(lastOffset, macroCallBody.currentOffset)
+                    val text = macroCallBody.originalText.substring(lastOffset, macroCallBody.currentOffset)
+
+                    // Wrap expressions in () to avoid problems related to operator precedence during expansion
```

Comment:
I chose to immediately add the () here instead of storing type and the unaltered text and adding them during the actual expansion. This way it requires a lot less changes to the code.